### DISCUSSION TITLE
Does anyone know how to check that the directory you mount a stratis file system to has been expanded?

### DISCUSSION THREAD

**Discussion by shanemhartley** [stratis-storage/project#497] [Q&A]
I am working through RHCSA objectives to sit for the certification. I've created a pool, and file system, mounted it, and expanded it.
Once it's expanded is there a way to validate that the directory you've mounted it on sees the storage expansion?

**Comment by jbaublitz** [comment]
Hi @shanebagel! `df` should report the increased size of the filesystem that's mounted.

**Comment by shanemhartley** [comment]
Do I have to do anything other than extend the pool for the filesystem to be extended too?
________________________________

**Comment by drckeefe** [comment]
Hello, shanebagel,
I believe by "expanded" you mean that the filesystem has grown.  Let me know if this is not what you are asking. 

If the file system expanded you will be able to see this change using the df and the command "stratis fs", which will show the fs size.

Stratis will grow  the filesystem automatically as long as there is enough physical space in the pool. 

Growing the pool (by adding additional data devices) is not need to grow the filesystem unless the pool has been depleted of physical 
space. 

What OS and version are you using and what is the version of Stratis?
'cat /etc/redhat-release'
'rpm -qa|grep stratis'

**Comment by shanemhartley** [comment]
RHEL 9.0
Stratisd - 2.4.2-3
Stratis-cli - 2.4.3-2

When I run df -h I get huge numbers for stratis file system size, it's saying the size is 1.0T, the amount used is 7.2g and amount available is 1017G
However the underlying pool only consists of 3 GB....
Also when I add to the pool those numbers don't seem to change.
________________________________

**Comment by drckeefe** [comment]
@shanebagel 
Stratis uses thin provisioning and by default creates a 1TB filesystem size, which in this case is much larger than your pool size. 
You will want to monitor the amount of free space in your pool 'stratis pool list' to make sure that you haven't run out of available space.
The df output displays the amount of used space from the filesystems perspective, though this doesn't always match with what has been stored in the pool.  The filesystem (XFS) is telling df that 7.2g is used because the filesystem would like to reserve the 7.2g for metadata space, in the event that the 1TB of space is used. 

In Stratis 3.0 https://stratis-storage.github.io/stratis-release-notes-3-0-0/
There was a feature enhancement to allow users to create filesystem of a specific size, since not all use cases required the default size.

**Comment by shanemhartley** [comment]
Thanks for the info!
________________________________
